spec/electric_slide/call_queue_spec.rb

```diff
@@ -138,6 +145,34 @@
           end
         end
       end
+
+      context "when the agent's and caller's calls are joined" do
+        before do
+          queued_call << Punchblock::Event::Joined.new(timestamp: connected_time)
+          agent_call << Punchblock::Event::Joined.new(timestamp: connected_time)
+        end
+
+        it "records the connection time in the :electric_slide_connected_at call variable on the queued call" do
+          expect(queued_call[:electric_slide_connected_at]).to eq(connected_time)
+        end
+      end
+
+      context "when the agent's and caller's calls are not joined" do
```

Comment:
Because the join events are not necessary conditions for all of them except that one. Plus [this scenario](https://github.com/neildecapia/electric_slide/blob/feature/queued-call-agent-variable/spec/electric_slide/call_queue_spec.rb#L160) should really be kept under "when the call ends" but the join event precondition there would have made setting up and writing this spec harder.